workflow/executor/executor.go

```diff
@@ -114,7 +114,13 @@ func (we *WorkflowExecutor) LoadArtifacts() error {
 	log.Infof("Start loading input artifacts...")
 
 	for _, art := range we.Template.Inputs.Artifacts {
+
 		log.Infof("Downloading artifact: %s", art.Name)
+
+		if !art.HasLocation() && art.Optional {
+			log.Warnf("Artifact doesn't have location. Artifact configured as an optional so, Artifact will be  ignored")
+			continue
+		}
```

Comment:
Minor improvement to give proper error, instead of failing  in InitDriver:

```
		if !art.HasLocation() {
		    art.Optional {
			log.Warnf("Artifact doesn't have location. Artifact configured as an optional so, Artifact will be  ignored")
			continue
		    }
		    return errors.New("required artifact %s not supplied", art.Name)
		}
```

Reply:
sure. done

Reply:
Why need this conditon？
if !art.HasLocation() {
}

We found that optional=true in Outputs Artifact don't need this condition。 
Or it should be if art.HasLocation() ?